File: design-pattern-java/app/src/test/java/design/pattern/java/patterns/creational/builder/BuilderTest.java
package design.pattern.java.patterns.creational.builder;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class BuilderTest {
  @Test
  public void itShouldCreateACarWithTheCorrectValuesWhenBuilderBuildIsCalled() {
    Integer doorsExpected = 2;
    String motorExpected = "6V";
    Float ltsExpected = 1.6f;
    Boolean acExpected = true;
    Boolean electronicsExpected = true;
    Car car = CarBuilder.create().setDoors(doorsExpected)
    .setMotor(motorExpected).setLts(ltsExpected).setAc(acExpected)
    .setElectronic(electronicsExpected).build();
    Assertions.assertEquals(doorsExpected, car.getDoors());
    Assertions.assertEquals(motorExpected, car.getMotor());
    Assertions.assertEquals(ltsExpected, car.getLts());
    Assertions.assertEquals(acExpected, car.getAc());
    Assertions.assertEquals(electronicsExpected, car.getElectronic());
  }
}

File: design-pattern-jvm/app/src/test/kotlin/design/pattern/kotlin/patterns/creational/builder/BuilderTest.kt
package design.pattern.kotlin.patterns.creational.builder

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class BuilderTestKt {
    @Test
    fun itShouldCreateACarWithTheCorrectValuesWhenBuilderBuildIsCalled() {
        val doorsExpected = 2
        val motorExpected = "6V"
        val ltsExpected = 1.6f
        val acExpected = true
        val electronicsExpected = true
        val (doors, motor, lts, ac, electronic) =
                CarBuilder.create()
                        .setDoors(doorsExpected)
                        .setMotor(motorExpected)
                        .setLts(ltsExpected)
                        .setAc(acExpected)
                        .setElectronic(electronicsExpected)
                        .build()
        Assertions.assertEquals(doorsExpected, doors)
        Assertions.assertEquals(motorExpected, motor)
        Assertions.assertEquals(ltsExpected, lts)
        Assertions.assertEquals(acExpected, ac)
        Assertions.assertEquals(electronicsExpected, electronic)
    }
}
